src/sets/signerSet.ts

```diff
@@ -99,381 +96,224 @@ class SignerSet {
-    for (const edgeKey of edges.keys()) {
-      const { childKey } = this.deconstructEdgeKey(edgeKey);
-      const existingSet = byChildMap.get(childKey) || new Set();
-      existingSet.add(edgeKey);
-      byChildMap.set(childKey, existingSet);
-    }
-    return byChildMap;
-  }
-
   /**
-   * getEdgesByParent returns a Map where each key is a parent vertex in the input edges set, and each value
-   * is a set of edges (as edge tuple strings) that contain the parentKey as the parent.
-   *
-   * For example, getEdgesByParent({"parentA,childB" => "hashC"}) => {"parentA" => Set(["parentA,childB"])}
+   * mergeCustodyRemoveAll moves all address in custodyAdds with a block number before the
+   * signer of the CustodyRemoveAll message to custodyRemoves
    *
-   * @param edges - a map of (parent, child) tuple strings to message hashes (i.e. _edgeAdds, _edgeRemoves, _edges)
+   * @param message - CustodyRemoveAll message
    */
-  private getEdgesByParent(edges: Map<string, string>) {
-    const byParentMap = new Map<string, Set<string>>();
-    for (const edgeKey of edges.keys()) {
-      const { parentKey } = this.deconstructEdgeKey(edgeKey);
-      const existingSet = byParentMap.get(parentKey) || new Set();
-      existingSet.add(edgeKey);
-      byParentMap.set(parentKey, existingSet);
-    }
-    return byParentMap;
-  }
+  private mergeCustodyRemoveAll(message: CustodyRemoveAll): Result<void, string> {
+    const sanitizedAddress = this.sanitizeKey(message.signer);
 
-  /**
-   * mergeSignerAdd calls add using the contents of the supplied SignerAdd message and adds message
-   * to the messages map if add was successful
-   *
-   * @param message - a SignerAdd message
-   */
-  private mergeSignerAdd(message: SignerAdd): Result<void, string> {
-    const parentKey = this.sanitizeKey(message.signer);
-    const childKey = this.sanitizeKey(message.data.body.childKey);
+    // If custodyAddress exists in custodyRemoves, no-op
+    if (this._custodyRemoves.has(sanitizedAddress)) return ok(undefined);
 
-    const res = this.add(parentKey, childKey, message.hash);
-    if (res.isErr()) return res;
+    // If custodyAddress does not exist in custodyAdds, fail
+    const custodyAddEvent = this._custodyAdds.get(sanitizedAddress);
+    if (!custodyAddEvent) return err('SignerSet.mergeCustodyRemoveAll: custodyAddress does not exist');
+
+    // For each custody address, remove if block number is before signer
+    for (const [address, addEvent] of this._custodyAdds) {
+      if (addEvent.blockNumber < custodyAddEvent.blockNumber) {
+        // Look through signerAdds
+        for (const [signerKey, signerAdd] of this._signerAdds) {
+          if (this.sanitizeKey(signerAdd.signer) === address) {
+            this._signerAdds.delete(signerKey);
+            this.emit('removeSigner', signerKey);
+          }
+        }
 
-    this._messages.set(message.hash, message);
-    return ok(undefined);
-  }
+        // Look through signerRemoves
```

Comment:
nit: would be helpful to elaborate on why for someone new to this

e..g `Remove all SignerRemoves that were authorized by a CustodyAddress that is now removed by this CustodyRemAll message`

Reply:
Yeah, I'll add more explanation in comments